pm/queue.go

```diff
@@ -110,22 +95,41 @@ func (q *ticketQueue) startQueueLoop() {
 	sub := q.blockSub(blockNums)
 	defer sub.Unsubscribe()
 
+ticketLoop:
 	for {
 		select {
 		case err := <-sub.Err():
 			if err != nil {
 				glog.Errorf("Block subscription error err=%v", err)
 			}
 		case latestBlock := <-blockNums:
-			numTickets := q.Length()
+			numTickets, err := q.Length()
+			if err != nil {
+				glog.Errorf("Error getting queue length err=%v", err)
+				continue
+			}
 			for i := 0; i < int(numTickets); i++ {
-				q.cond.L.Lock()
-				nextTicket := q.queue[0]
-				q.cond.L.Unlock()
-				q.removeHead()
+				nextTicket, err := q.pop()
+				if err != nil {
+					glog.Errorf("Unable to pop ticket from queue err=%v", err)
+					continue ticketLoop
+				}
+				if nextTicket == nil {
+					continue ticketLoop
+				}
 				if nextTicket.ParamsExpirationBlock.Cmp(latestBlock) <= 0 {
+					resCh := make(chan struct {
+						txHash ethcommon.Hash
+						err    error
+					})
 					select {
-					case q.redeemable <- nextTicket:
+					case q.redeemable <- &redemption{nextTicket, resCh}:
+						res := <-resCh
+						if res.err != nil {
+							glog.Error(err)
+							continue
+						}
+						q.store.MarkWinningTicketRedeemed(nextTicket, res.txHash)
```

Comment:
Need to handle error

Reply:
fixed in d2a979a8cc8d70cfd436c82a5c06da150c8e9b0c